File: Semester 1/Class06_Computer_Contest_Lv_2_Notes_and_Homework_Oct_16-17/lvl2_hw_sol/lvl2_triangles.py
import sys
input = sys.stdin.readline
n, ans = int(input()), 0
x, y = [], []

for i in range(n):
    px, py = map(int, input().split())
    x.append(px)
    y.append(py)

for i in range(n):
    for j in range(n):
        for k in range(n):
            if x[i] == x[j] and y[i] == y[k]:
                ans = max(ans, abs(y[i] - y[j]) * abs(x[i] - x[k]))

print(ans)

File: Semester 1/Class06_Computer_Contest_Lv_2_Notes_and_Homework_Oct_16-17/lvl2_hw_sol/lvl2_triangles.cpp
#include <bits/stdc++.h>
using namespace std;
#define f first
#define s second
typedef pair<int, int> pi;
int N, ans; vector<pi> p;
int main(){
    scanf("%d", &N);
    for(int i=0, x, y; i<N; i++){
        scanf("%d %d", &x, &y);
        p.push_back({x, y});
    }
    for(int i=0; i<N; i++)
        for(int j=0; j<N; j++)
            for(int k=0; k<N; k++)
                if(p[i].f == p[j].f && p[i].s == p[k].s)
                    ans = max(ans, abs(p[i].s - p[j].s)*abs(p[i].f - p[k].f));
    printf("%d\n", ans);
}